app/uk/gov/hmrc/nationaldutyrepaymentcenter/models/requests/EISCreateCaseRequest.scala

```diff
@@ -61,16 +61,46 @@ object EISCreateCaseRequest {
     def from(request: CreateClaimRequest): Content = {
       Content(
         ClaimDetails = request.Content.ClaimDetails,
-        AgentDetails = request.Content.AgentDetails match {
-          case Some(result) => Some(result)
+        AgentDetails = request.Content.AgentDetails.isDefined match {
+          case true  => Some(getAgentUserDetails(request))
           case _ => None
         },
-        ImporterDetails = request.Content.ImporterDetails,
+        ImporterDetails = getImporterDetails(request),
         BankDetails = request.Content.BankDetails,
         DutyTypeTaxDetails = request.Content.DutyTypeTaxDetails,
         DocumentList = request.Content.DocumentList
       )
     }
+
+    def getImporterDetails( request: CreateClaimRequest) : EISUserDetails = {
+      val name = UserName(request.Content.ImporterDetails.Name.firstName,
+          request.Content.ImporterDetails.Name.lastName)
+
+      val fullName:String = (name.firstName+ name.lastName).mkString("")
+
+      EISUserDetails(
+        IsVATRegistered = request.Content.ImporterDetails.IsVATRegistered,
+        EORI = request.Content.ImporterDetails.EORI,
+        Name =  fullName,
+        Address = request.Content.ImporterDetails.Address
+      )
+    }
+
+
+    def getAgentUserDetails(request: CreateClaimRequest): EISUserDetails =  {
+      val name = UserName(request.Content.AgentDetails.get.Name.firstName,
+          request.Content.ImporterDetails.Name.lastName)
+
+      val fullName:String = (name.firstName+ name.lastName).mkString("")
```

Comment:
```suggestion
      val fullName:String = (name.firstName + name.lastName).mkString(" ")
```

Reply:
![image](https://user-images.githubusercontent.com/58818306/104911468-cf17dc00-5982-11eb-9a68-da7d09e77a4f.png)
@lmugnai  implementing **mkString(" ")** will result within the attached screenshot

Reply:
OK, thanks for checking. Still a bit unsure, "The mkString() method is utilized to display all the elements of the list in a string along with a separator". How is the space between the 2 being inserted? Also, is firstName (and lastName) a List?